catalog/app/components/Pagination/index.js

```diff
@@ -0,0 +1,181 @@
+import messages from './messages';
+
+
+const UNIT_SIZE = '24px';
+
+
+const Controls = styled.div`
+  display: flex;
+  height: ${UNIT_SIZE};
+  justify-content: space-between;
+  margin-top: 16px;
+`;
+
+const Chevron = compose(
+  setPropTypes({
+    direction: PT.oneOf(['left', 'right']).isRequired,
+  }),
+  withProps(({ direction }) => ({
+    children: <MIcon>{`chevron_${direction}`}</MIcon>,
+    style: {
+      height: UNIT_SIZE,
+      padding: 0,
+      width: UNIT_SIZE,
+    },
+  })),
+)(IconButton);
+
+const Pages = styled.span`
+  font-weight: bold;
+`;
+
+const Dim = styled.span`
+  color: ${grey500};
+`;
+
+const InlineDropDown = withProps({
+  style: {
+    height: UNIT_SIZE,
+    verticalAlign: 'bottom',
+  },
+  iconStyle: {
+    border: 'none',
+    height: '100%',
+    padding: 0,
+    right: 0,
+    top: 0,
+    width: UNIT_SIZE,
+  },
+  underlineStyle: {
+    margin: 0,
+  },
+  labelStyle: {
+    color: 'inherit',
+    height: UNIT_SIZE,
+    lineHeight: UNIT_SIZE,
+    paddingLeft: 0,
+    paddingRight: UNIT_SIZE,
+  },
+})(DropDownMenu);
+
+const renderPerPageOption = (v) => (
+  <MenuItem
+    value={v}
+    key={v}
+    primaryText={v === 'all' ? <FM {...messages.all} /> : v}
+  />
+);
+
+const PerPage = PT.oneOfType([PT.number, PT.oneOf(['all'])]);
+
+export default compose(
+  setPropTypes({
+    items: PT.array.isRequired,
+    page: PT.number.isRequired, // 1-based
+    onPageChange: PT.func.isRequired,
+    perPageOptions: PT.arrayOf(PerPage),
+    perPage: PerPage,
+    onPerPageChange: PT.func.isRequired,
+    children: PT.func.isRequired,
+  }),
+  defaultProps({
+    perPageOptions: DEFAULT_PER_PAGE_OPTIONS,
+    perPage: DEFAULT_PER_PAGE,
+  }),
+  pure,
+  withProps(({ page, perPage, perPageOptions, items }) => {
+    assert(perPageOptions.includes(perPage),
+      `Pagination: invalid perPage: ${perPage}`);
+
+    if (perPage === 'all' || items.length === 0) {
+      return {
+        pages: 1,
+        pageItems: items,
+      };
+    }
+
+    assert(perPage > 0, 'Pagination: perPage must be a positive integer');
+    assert(page >= 1, 'Pagination: page must be a natural number');
+
+    const offset = (page - 1) * perPage;
+    assert(offset < items.length,
+      `Pagination: offset (${offset}) exceeds number of items (${items.length})`);
+
+    return {
+      pages: Math.ceil(items.length / perPage),
+      pageItems: items.slice(offset, offset + perPage),
+    };
+  }),
+  withHandlers({
+    nextPage: ({ page, onPageChange }) => () => onPageChange(page + 1),
+    prevPage: ({ page, onPageChange }) => () => onPageChange(page - 1),
+    handlePerPageChange: ({ onPerPageChange }) => (_e, _i, value) => {
+      onPerPageChange(value);
+    },
+  }),
+  setDisplayName('Pagination'),
+)(({
+  items,
+  page,
+  pageItems,
+  pages,
+  nextPage,
+  prevPage,
+  perPageOptions,
+  perPage,
+  handlePerPageChange,
+  children,
+}) => (
+  <div>
+    {children({ items: pageItems })}
+    {!!items.length &&
```

Comment:
what is intended here? ternary might be clearer

Reply:
The intention is to only show the controls if items length is non-zero. `!!` is a shorthand for boolean conversion (roughly equivalent to `Boolean(items.length)`). If we omit it, literal `0` is rendered when the length is zero, but if we use it, react renders nothing as intended. Ternary would be a little more noisy, something like this:

```javascript
{items.length === 0 ? null :
  ...
}
```